internal/xds/translator/oidc.go

```diff
@@ -228,9 +230,48 @@ func oauth2Config(oidc *ir.OIDC) (*oauth2v3.OAuth2, error) {
 		}
 		oauth2.Config.RetryPolicy = rp
 	}
+
+	if oidc.PassThroughAuthHeader {
+		oauth2.Config.PassThroughMatcher = buildHeaderMatchers(securityFeatures.JWT)
+	}
+
 	return oauth2, nil
 }
 
+func buildHeaderMatchers(jwt *ir.JWT) []*routev3.HeaderMatcher {
+	// Bypass OIDC if a header that will be handled by JWT is passed.
+	passThroughMatchers := make([]*routev3.HeaderMatcher, 0, len(jwt.Providers))
+
+	for _, provider := range jwt.Providers {
+		if provider.ExtractFrom == nil {
```

Comment:
Shouldn´t we do here:
`provider.ExtractFrom == nil || provider.ExtractFrom.Headers == nil`

Because ExtractFrom might contain the id-token-cookie from the OIDC Filter

Reply:
looks like this has been addressed in the preceding Gateway API layer
```

		hasValidJwtExtractor := false
		for _, provider := range jwt.Providers {
			// When ExtractFrom is not specified it falls back to looking at the "Authorization: Bearer ..." header
			if provider.ExtractFrom == nil || len(provider.ExtractFrom.Headers) > 0 {
				hasValidJwtExtractor = true
				break
			}
		}
		if !hasValidJwtExtractor {
			return errors.New("the OIDC.PassThroughAuthHeader setting must be used in conjunction with a JWT provider that is configured to read from a header")
		}
```

Reply:
oh nvm, I understand Dennis's question here, `if provider.ExtractFrom == nil` will not kick in if there's a cookie containing a JWT that has an authentication rule
so maybe lets add
```
provider.ExtractFrom == nil || provider.ExtractFrom.Headers == nil
```

to solve this @StephenRobin

Reply:
I may not understand this correctly, but if we use `provider.ExtractFrom == nil || provider.ExtractFrom.Headers == nil` here,  and a JWT token cookie is in the request, `Authorization` will be added to the passthroughMatchers, which shouldn't because `Authorization` is only used by default when `provider.ExtractFrom == nil`.